### ISSUE TITLE
Overlap between Readable and Writable streams

### ISSUE THREAD

**Issue by npenin** [microsoft/TypeScript#35326]
I am not sure whether the bug should be reported here or in Definitelytyped, but I assume a new syntaxic sugar is required to cover this kind of cases.

**TypeScript Version:**  3.7.2

**Search Terms:** overlap readable writable stream multiple inheritance

**Code**

```ts
import { Readable, Writable, Duplex } from 'stream';
var duplex = new Duplex();
var inStream: Readable=duplex;
var outStream: Writable=duplex;

if (inStream !== outStream)
{
    //there should be no compilation error
}
```

**Expected behavior:** It should not claim that Readable and Writable are not overlapping while it clearly is the case.

**Actual behavior:** it reports that the condition will always be true as there is no overlap between Readable and Writable, which, according to the little code sample is clearly wrong.

**Playground Link:** http://www.typescriptlang.org/play/index.html#code/JYWwDg9gTgLgBAbzgJQKYEMAm6BGAbVAGjgHUpgZcDiARAVzAIA84BfOAMyghDgHIAzjCgYQfANwAoAG7oocTA2ZwAvHAB2qAO5x6jVEwAUASimz5wdQGVhogFwoM2fKhWL9TM3LgQ6MGyLoIA5kFFSu7sxSksAccIaWAaJwAIQqar7+tkHGkgiScIVwAPTFMAAWqCJwAuW+eJhwOKgaEHAAxjxgwHjoMMAQ6nBV3FCSrEA

**Related Issues:**

**Comment by RyanCavanaugh** [comment]
This is the intended behavior. In general for any two types, unless there is some reason to believe otherwise, they are assumed to be disjoint and thus a likely error to compare with `===` / `!==`.

IOW TS has no way to distinguish `inStream === outStream` vs `myCat === myMortgage` -- the existence of something that is at once both a Cat and a Mortgage is not impossible per se, but is assumed not to happen.

**Comment by npenin** [comment]
ok, so the only workaround to that comparison would be change the type of inStream to be something like `var inStream: Readable & Writable | Readable = duplex;`, right ?
